include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
@awulkiew That is a different thing.You mentioned it is a general problem in BG. I reacted on that: it is not a general problem in BG.

UDT are a different thing - if you cannot assume they have a default constructor, you can also not assume the presence of a constructor T(0). Therefore we selected 7 years ago the construction of T(): it is well defined.

Eehm, as far as I know I never wrote the construct as you mention for quarter - this should IMO *not* be done. It should read like this: return_type const quarter = 0.25; We should fix that.